buildAndReleaseTask/pulumi.ts

```diff
@@ -102,26 +103,37 @@ export async function runPulumi() {
         tl.debug(tl.loc("Debug_PrintingVersion"));
         await tl.exec(toolPath, "version");
 
+        const agentEnvVars = tryGetEnvVars();
+        const loginEnvVars = {
+            ...agentEnvVars,
+        };
+        
+        const azureStorageContainer = agentEnvVars["AZURE_STORAGE_CONTAINER"];
+
         // Login and then run the command.
         tl.debug(tl.loc("Debug_Login"));
         // For backward compatibility, also check for `pulumi.access.token`
         // and manually set the access token env var for the login command.
         const pulumiAccessToken =
             tl.getVariable("pulumi.access.token") ||
+            // `getVariable` will automatically prepend the SECRET_ prefix if it finds
+            // it in the build environment's secret vault.
             tl.getVariable("PULUMI_ACCESS_TOKEN");
-        if (!pulumiAccessToken) {
+        if (!azureStorageContainer && !pulumiAccessToken) {
             tl.setResult(tl.TaskResult.Failed, tl.loc("PulumiAccessTokenNotFound"));
```

Comment:
Can you please change the localization variable to `PulumiLoginUndetermined`? You will then need to add that to `task.json`. You can also remove `PulumiAccessTokenNotFound` from `task.json` while you are in that file anyway since it is not used anywhere else other than here.

Reply:
Yes, will sent you and update with this and a new error message appropriate for this new situation.

Reply:
Hmm this looks like it wasn't done? Can you please make this change as well? (Unresolving this conversation.)

Reply:
aaaa.. sorry, missed to commit some files, push them now, hopefully it's all there.